x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -556,21 +556,38 @@ protected FileChunk nextChunkRequest(StoreFileMetadata md) {
                 protected void executeChunkRequest(FileChunk request, ActionListener<Void> listener) {
                     remoteClient.execute(GetCcrRestoreFileChunkAction.INSTANCE,
                         new GetCcrRestoreFileChunkRequest(node, sessionUUID, request.md.name(), request.bytesRequested),
-                            ListenerTimeouts.wrapWithTimeout(threadPool, ActionListener.delegateFailure(listener, (l, r) -> {
-                                r.incRef();
-                                threadPool.generic().execute(new ActionRunnable<>(listener) {
-                                    @Override
-                                    protected void doRun() throws Exception {
-                                        writeFileChunk(request.md, r);
-                                        listener.onResponse(null);
-                                    }
-
-                                    @Override
-                                    public void onAfter() {
-                                        r.decRef();
-                                    }
-                                });
-                            }), ccrSettings.getRecoveryActionTimeout(), ThreadPool.Names.GENERIC, GetCcrRestoreFileChunkAction.NAME));
+                            ListenerTimeouts.wrapWithTimeout(threadPool, new ActionListener<>() {
+                                @Override
+                                public void onResponse(
+                                        GetCcrRestoreFileChunkAction.GetCcrRestoreFileChunkResponse getCcrRestoreFileChunkResponse) {
+                                    getCcrRestoreFileChunkResponse.incRef();
+                                    threadPool.generic().execute(new ActionRunnable<>(listener) {
+                                        @Override
+                                        protected void doRun() throws Exception {
+                                            writeFileChunk(request.md, getCcrRestoreFileChunkResponse);
+                                            listener.onResponse(null);
+                                        }
+
+                                        @Override
+                                        public void onAfter() {
+                                            getCcrRestoreFileChunkResponse.decRef();
+                                        }
+                                    });
+                                }
+
+                                @Override
+                                public void onFailure(Exception e) {
+                                    threadPool.generic().execute(() -> {
+                                        try {
+                                            listener.onFailure(e);
+                                        } catch (Exception ex) {
```

Comment:
This whole thing is kind of verbose but since we need to fork-off after incrementing the ref there wasn't a nice way of doing this via `ThreadedActionListener` and this recreates the old behavior of that class pretty much exactly and with a more meaningful log message.
(we definitely have to catch here, the callback does blocking transport requests that will fail at some rate)

Reply:
This was originally broken here: https://github.com/elastic/elasticsearch/pull/65921/files#diff-99f9df7f4e940afb7f2885ff7fddebe882f88345db74519a06744b019afab8b7R559